### PR TITLE
wolfHSM support for ML-DSA

### PR DESCRIPTION
**Pull request by bigbrett**
- Adds wolfHSM offload for ML-DSA
- Consolidates AURIX helper scripts into `wbaurixtool.sh`, a one-stop-shop configuration tool that handles the manual configuration steps that aren't handled by AURIX IDE
- Various AURIX project configuration improvements

~~Note that this requires updating wolfSSL to include https://github.com/wolfSSL/wolfssl/pull/8177 and https://github.com/wolfSSL/wolfssl/pull/8229. If we need to wait until the required changes are incorporated in a release, let me know and this can remain a draft PR~~

### CODE REVIEW

**Comment by bigbrett** [comment]
Converting to draft until @danielinux merges #521

**Comment by bigbrett** [comment]
rebased after #521 was merged, re-tested on Sim + AURIX. Ready for review

File: tools/keytools/keygen.c [hunk 1]
```diff
@@ -1086,8 +1086,45 @@ static void keygen_ml_dsa(const char *priv_fname, uint32_t id_mask)
     fwrite(pub, pub_len, 1, fpriv);
     fclose(fpriv);
 
-    keystore_add(KEYGEN_ML_DSA, pub, pub_len,
-                 priv_fname, id_mask);
+    if (exportPubKey) {
+        if (saveAsDer) {
+            /* Export public key in DER format */
+            uint8_t pubDer[
+            #if ML_DSA_LEVEL == 2
```
Comment (dgarske) [inline tools/keytools/keygen.c:1093]: These come from run-time arguments now. You might need to rebase and fix this.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: tools/keytools/keygen.c (hunk 1 above)
Comment (bigbrett) [reply tools/keytools/keygen.c]: oops. Already had rebased but forgot to change this line. Fixed in latest commit to switch on `ml_dsa_level`

File: tools/keytools/keygen.c [hunk 2]
```diff
@@ -1086,8 +1086,64 @@ static void keygen_ml_dsa(const char *priv_fname, uint32_t id_mask)
...
+                    break;
+                case WC_ML_DSA_87:
+                    pubDerSz = ML_DSA_LEVEL5_PUB_KEY_DER_SIZE;
+                    break;
+                default:
+                    fprintf(stderr, "Error: Unsupported ML DSA level\n");
+                    exit(1);
+                    break;
+            }
+            pubDer = malloc(pubDerSz);
```
Comment (dgarske) [inline tools/keytools/keygen.c:1112]: Cast `(uint8_t*)`

File: tools/keytools/keygen.c [hunk 3]
```diff
@@ -1086,8 +1086,64 @@ static void keygen_ml_dsa(const char *priv_fname, uint32_t id_mask)
...
+                        "Error: Failed to allocate memory for DER export\n");
+                exit(1);
+            }
+
+            /* Export public key in DER format */
+
+            pubOutLen = wc_Dilithium_PublicKeyToDer(&key, pubDer, pubDerSz,
+                                                    WITH_ALG_SPKI);
+            if (pubOutLen < 0) {
+                fprintf(stderr, "Unable to export public key to DER, ret=%d\n",
```
Comment (dgarske) [inline tools/keytools/keygen.c:1124]: Do you want to free `pubDer`?

**Review by dgarske — DISMISSED** [review]

File: tools/keytools/keygen.c (hunk 3 above)
Comment (bigbrett) [reply tools/keytools/keygen.c:1124]: we immediately `exit(1)` on any failure, so didn't free. Just following the paradigm elsewhere in `keygen.c`.

File: tools/keytools/keygen.c (hunk 2 above)
Comment (bigbrett) [reply tools/keytools/keygen.c]: added in latest commit. Didn't realize this was in our coding standard, but indeed it is. Just of curiosity why do we do this? For C++ compatibility?

Comment (dgarske) [reply tools/keytools/keygen.c]: It is in our coding standards `Cast XMALLOC. byte* var = (byte*)XMALLOC(size, NULL, DYN);`. When building the tool in Windows or with g++ compiler or with the right compiler flags you will see this as a warning.

**Review by dgarske — APPROVED** [review]
Looks good. over to @danielinux

File: include/user_settings.h [hunk 4]
```diff
@@ -251,15 +251,17 @@ extern int tolower(int c);
     /* Wolfcrypt builds ML-DSA (dilithium) to the FIPS 204 final
      * standard by default. Uncomment this if you want the draft
      * version instead. */
-    #if 0
-    #define WOLFSSL_DILITHIUM_FIPS204_DRAFT
-    #endif
+#   if 0
```
Comment (danielinux) [inline include/user_settings.h:254]: Remove block?

**Review by danielinux — COMMENTED** [review]
nit: remove dead config block

File: include/user_settings.h (hunk 4 above)
Comment (danielinux) [reply include/user_settings.h:254]: (not related to this commit), taking back)

**Pull request merged**